Task configure sa: 'Add image pull secrets to a service account' sect…

scoulomb opened the pull request:
Improvement proposal to show:
- editing yaml file is an alternative of path
- `spec.imagePullSecrets` field added if in namespace but also using correct sa  (line 264)

scoulomb commented:
/assign @jimangel

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
````diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
-```shell
-kubectl get secrets myregistrykey
-```
+- Create an imagePullSecret, as described [here](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
````
nit:
```suggestion
- Create an imagePullSecret, as described in [Specifying ImagePullSecrets on a Pod](/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod).
```

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
````diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
-myregistrykey    kubernetes.io/.dockerconfigjson   1       1d
-```
+- Verify it has been created.
+   ```shell script
````
nit: `shell script` should be just `shell`

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
````diff
@@ -183,27 +183,38 @@ The content of `token` is elided here.
 ```
 
-Interactive version requires manual edit:
+Alternatively we can use `kubectl edit` or edit manually yaml files as shown below:
````
This wording would be more idiomatic:
```suggestion
You can instead use `kubectl edit`, or manually edit the YAML manifests as shown below:
```

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
```diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
-Now, any new pods created in the current namespace will have this added to their spec:
+### Verify imagePullSecrets was added to pod spec
+
+Now, any new pods created in the current namespace and using default service account will have `spec.imagePullSecrets` field added:
```
How about this wording:
```suggestion
Now, when a new Pod is created in the current namespace and using the default ServiceAccount, the new Pod has its  `spec.imagePullSecrets` field set automatically:
```

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
`````diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
+kubectl get pod nginx -o yaml | grep -A 1 imagePullSecrets
+````
+
+Output is
`````
nit:
```suggestion
The output is:
```

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
`````diff
@@ -248,10 +259,18 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
+
+````shell
+kubectl run nginx --image=nginx --restart=Never
+kubectl get pod nginx -o yaml | grep -A 1 imagePullSecrets
`````
This could also use a [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) query. How about taking that option (rather than `grep`)?

scoulomb replied:
Yes, it is much better!

scoulomb commented:
> @scoulomb I've added some feedback. Anything marked as a “nit” means that I think this is a good idea to fix, but also that this is fine to merge without addressing. Also, the questions genuinely are questions rather than mandates.
> 
> I hope you find these suggestions useful. If you would like to ask questions about this feedback please feel free to do that.

@sftim Thanks a lot for all your suggestions! it was very useful. I updated.

scoulomb commented:
/assign @jimangel

sftim commented on content/en/docs/tasks/configure-pod-container/configure-service-account.md:
````diff
@@ -248,12 +259,19 @@ Finally replace the serviceaccount with the new updated `sa.yaml` file
+
+The output is:
+
+```shell
````
nit: this is sample output, not shell code

scoulomb commented:
> There's a snag but it's not a big one. @scoulomb if you wanted to fix that please do feel free to send in a new PR.
> 
> /lgtm
> /approve

@sftim thanks I will fix this
